code/datums/actions/mobs/charge.dm

```diff
@@ -156,17 +156,23 @@
 			SSexplosions.med_mov_atom += target
 
 	INVOKE_ASYNC(src, PROC_REF(DestroySurroundings), source)
-	hit_target(source, target, charge_damage)
-
-/datum/action/cooldown/mob_cooldown/charge/proc/hit_target(atom/movable/source, atom/target, damage_dealt)
-	if(!isliving(target))
-		return
-	var/mob/living/living_target = target
-	living_target.visible_message("<span class='danger'>[source] slams into [living_target]!</span>", "<span class='userdanger'>[source] tramples you into the ground!</span>")
-	source.forceMove(get_turf(living_target))
-	living_target.apply_damage(damage_dealt, BRUTE, wound_bonus = CANT_WOUND)
-	playsound(get_turf(living_target), 'sound/effects/meteorimpact.ogg', 100, TRUE)
-	shake_camera(living_target, 4, 3)
+	try_hit_target(source, target)
+
+/// Attempt to hit someone with our charge
+/datum/action/cooldown/mob_cooldown/charge/proc/try_hit_target(atom/movable/source, atom/target)
+	if (can_hit_target(source, target))
+		hit_target(source, target, charge_damage)
+
+/// Returns true if we're allowed to charge into this target
+/datum/action/cooldown/mob_cooldown/charge/proc/can_hit_target(atom/movable/source, atom/target)
+	return isliving(target)
+
+/// Actually hit someone
+/datum/action/cooldown/mob_cooldown/charge/proc/hit_target(atom/movable/source, mob/living/target, damage_dealt)
+	target.visible_message("<span class='danger'>[source] slams into [target]!</span>", "<span class='userdanger'>[source] tramples you into the ground!</span>")
```

Comment:
use `span_danger()` and `span_userdanger()` now that we're cleaning this stuff up also multiline